odl/solvers/nonsmooth/douglas_rachford.py

```diff
@@ -118,21 +131,28 @@ def douglas_rachford_pd(x, f, g, L, tau, sigma, niter,
     composite and parallel-sum type monotone operators*. SIAM Journal
     on Optimization, 23.4 (2013), pp 2541--2565.
     """
-    # Problem size
-    m = len(L)
-
     # Validate input
+    m = len(L)
     if not all(isinstance(op, Operator) for op in L):
         raise ValueError('`L` not a sequence of operators')
     if not all(op.is_linear for op in L):
         raise ValueError('not all operators in `L` are linear')
     if not all(x in op.domain for op in L):
         raise ValueError('`x` not in the domain of all operators')
-    if len(sigma) != m:
-        raise ValueError('len(sigma) != len(L)')
     if len(g) != m:
         raise ValueError('len(prox_cc_g) != len(L)')
 
+    if tau is None or sigma is None:
+        tau_default, sigma_default = douglas_rachford_pd_stepsize(L, tau)
```

Comment:
We want to support the case `sigma` given but not `tau`. I'll implement this

Reply:
What's missing?

Reply:
Basically there are 4 use cases:

None, None
tau, None
None, Sigma
tau, sigma

To the best of my understanding, this version only handles these ones

None, None
tau, None
tau, sigma

right?

Reply:
In the "None, sigma" case the method is run and you get the defaults (`tau=None` is OK for the helper), and then you set `tau` to its default and `sigma` is skipped. 
So to me it seems all cases are handled, or do I have this backwards?

Reply:
Lets say I pick sigma = [HUGE NUMBER], then this would still pick tau=[AVERAGE NUMBER], instead of tau=[TINY NUMBER], right?

So sure, it is handled, but I don't think we would have convergence?

Reply:
Sure, it's always the case that `sigma` depends on `tau`, not the other way around. Programmatically speaking, all cases are handled, but one of them incorrectly (not guaranteeing convergence). In that sense you're right.
I guess you'd take something like `2 / sum(sigma)` for that case.